samples/drivers/i2s/output/src/main.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/i2s.h>
+#include <zephyr/sys/iterable_sections.h>
+
+#define SAMPLE_NO 64
+
+/* The data represent a sine wave */
+static int16_t data[SAMPLE_NO] = {
+	  3211,   6392,   9511,  12539,  15446,  18204,  20787,  23169,
+	 25329,  27244,  28897,  30272,  31356,  32137,  32609,  32767,
+	 32609,  32137,  31356,  30272,  28897,  27244,  25329,  23169,
+	 20787,  18204,  15446,  12539,   9511,   6392,   3211,      0,
+	 -3212,  -6393,  -9512, -12540, -15447, -18205, -20788, -23170,
+	-25330, -27245, -28898, -30273, -31357, -32138, -32610, -32767,
+	-32610, -32138, -31357, -30273, -28898, -27245, -25330, -23170,
+	-20788, -18205, -15447, -12540,  -9512,  -6393,  -3212,     -1,
+};
+
+/* Fill buffer with sine wave on left channel, and sine wave shifted by
+ * 90 degrees on right channel. "att" represents a power of two to attenuate
+ * the samples by
+ */
+static void fill_buf(int16_t *tx_block, int att)
+{
+	int r_idx;
+
+	for (int i = 0; i < SAMPLE_NO; i++) {
+		/* Left channel is sine wave */
+		tx_block[2 * i] = data[i] / (1 << att);
+		/* Right channel is same sine wave, shifted by 90 degrees */
+		r_idx = (i + (ARRAY_SIZE(data) / 4)) % ARRAY_SIZE(data);
+		tx_block[2 * i + 1] = data[r_idx] / (1 << att);
+	}
+}
+
+#define NUM_BLOCKS 20
+#define BLOCK_SIZE (2 * sizeof(data))
+
+#ifdef CONFIG_NOCACHE_MEMORY
+	#define MEM_SLAB_CACHE_ATTR __nocache
+#else
+	#define MEM_SLAB_CACHE_ATTR
+#endif /* CONFIG_NOCACHE_MEMORY */
+
+const struct device *dev_i2s = DEVICE_DT_GET(DT_ALIAS(i2s_tx));
+
+char MEM_SLAB_CACHE_ATTR __aligned(WB_UP(32))
+	_k_mem_slab_buf_tx_0_mem_slab[(NUM_BLOCKS) * WB_UP(BLOCK_SIZE)];
+
+STRUCT_SECTION_ITERABLE(k_mem_slab, tx_0_mem_slab) =
+	Z_MEM_SLAB_INITIALIZER(tx_0_mem_slab, _k_mem_slab_buf_tx_0_mem_slab,
+				WB_UP(BLOCK_SIZE), NUM_BLOCKS);
```

Comment:
Couldn't `K_MEM_SLAB_DEFINE_STATIC` be used instead?

Reply:
We can't here unfortunately- due to the need to place the memory slab into noncached memory on some platforms.